.github/workflows/integration_test.yaml

```diff
@@ -160,199 +175,135 @@ on:
       zap-rules-file-name:
         description: Rules file to ignore any alerts from the ZAP scan
         type: string
-    outputs:
-      images:
-        description: Pushed images
-        value: ${{ jobs.all-images.outputs.images }}
 
 env:
   REGISTRY: ghcr.io
   OWNER: ${{ github.repository_owner }}
 
-concurrency:
-  group: operator-workflows-${{ github.workflow }}-integration-tests-${{ github.ref }}-${{ inputs.test-tox-env }}
-  cancel-in-progress: true
-
 jobs:
-  get-runner-image:
-    name: Get runner image
-    uses: ./.github/workflows/get_runner_image.yaml
-    with:
-      working-directory: ${{ inputs.working-directory }}
-  build-images:
-    name: Build image
-    uses: ./.github/workflows/build_images.yaml
-    needs: get-runner-image
-    with:
-      owner: ${{ github.repository_owner }}
-      registry: ghcr.io
-      runs-on: ${{ needs.get-runner-image.outputs.runs-on }}
-      trivy-image-config: ${{ inputs.trivy-image-config }}
-      working-directory: ${{ inputs.working-directory }}
-      build-args: ${{ inputs.image-build-args }}
-  build-rocks:
-    name: Build rock
-    uses: ./.github/workflows/build_rocks.yaml
-    needs: get-runner-image
-    with:
-      owner: ${{ github.repository_owner }}
-      registry: ghcr.io
-      runs-on: ${{ needs.get-runner-image.outputs.runs-on }}
-      trivy-image-config: ${{ inputs.trivy-image-config }}
-      working-directory: ${{ inputs.working-directory }}
-      rockcraft-channel: ${{ inputs.rockcraft-channel }}
-      rockcraft-repository: ${{ inputs.rockcraft-repository }}
-      rockcraft-ref: ${{ inputs.rockcraft-ref }}
-  all-images:
-    name: Get rocks or Docker images
-    needs: [ build-images, build-rocks ]
+  plan:
+    name: Plan
     runs-on: ubuntu-latest
     outputs:
-      images: ${{ env.IMAGES }}
+      plan: ${{ steps.plan.outputs.plan }}
     steps:
-      - name: Get rocks or fallback to Docker images
-        run: |
-          if [ ${{ needs.build-rocks.outputs.images != '[]' }} = true ]; then
-            IMAGES='${{ needs.build-rocks.outputs.images }}'
-          else
-            IMAGES='${{ needs.build-images.outputs.images }}'
-          fi
-          echo "IMAGES=$IMAGES" >> $GITHUB_ENV
-  build-charm:
-    name: Build and push charm
-    needs: get-runner-image
-    runs-on: ${{ needs.get-runner-image.outputs.runs-on }}
-    outputs:
-      charm-file: ${{ env.CHARM_FILE }}
+      - uses: actions/checkout@v4.1.1
+      - uses: canonical/operator-workflows/internal/plan@better-build
```

Comment:
Indeed, but setting this to `@main` in the pull request would cause the tests to fail, as the actions do not yet exist in the main branch at this point. Therefore, I will need to update all the references in the final step of this pull request, just before merging.

Reply:
Yes, that is why I meant **after merging** .